### ISSUE TITLE
Cookie.set not storing cookie when path is passed

### ISSUE THREAD

**Issue by Shooshte** [js-cookie/js-cookie#506]
using: `"js-cookie": "^2.2.0"`

```
import Cookie from 'js-cookie';

Cookie.set('EV-Credentials', 'blah', {
      path: '/system',
      expires: 7,
    });

const token = Cookie.get('EV-Credentials', { path: '/system' });
console.log(token);
```

Console log returns undefined. 

If I do not pass the path parameter into `Cookie.set` or pass an empty string, it defaults to `/` for the cookie path and everything works correctly.

**Comment by carhartl** [comment]
First of all, for technical reasons it is not possible to get a particular cookie by path. You can only get a cookie, and the client will return the cookie it has access to at the current location. Meaning if you have a cookie with the same name at both `/` and `/foo` one cookie is going to shadow the other one. 

Similarly, while you can write a cookie at any path from any location, when you’re currently not at `/system` you won’t be able to retrieve this cookie.

Please make sure that’s not the issue. 

Long story short, you can only retrieve a cookie by key: `Cookie.get('EV-Credentials')` and when doing so you must have access to it taking into account the domain and path settings.

**Comment by Shooshte** [comment]
@carhartl
Thanks for the clarification using `Cookie.get`  👍 

Maybe I misunderstood the usage of the `path` param?

I am using `js-cookie` inside a client application. My use case for setting path was to limit the cookie to only be sent when XML requests are sent to a certain path.

So basically, what the [MDN cookie page](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies) specifies:
```
Path indicates a URL path that must exist in the requested URL in order to send the Cookie header. The %x2F ("/") character is considered a directory separator, and subdirectories will match as well.

For example, if Path=/docs is set, these paths will match:

/docs
/docs/Web/
/docs/Web/HTTP
```

If I understand correctly what you are saying is that if my web app location (the browser URL) is not pointing to `/system` - the path from the example, I will not be able to access the cookie?

**Comment by carhartl** [comment]
> If I understand correctly what you are saying is that if my web app location (the browser URL) is not pointing to /system - the path from the example, I will not be able to access the cookie?

Yes!

**Comment by Shooshte** [comment]
Thanks for the clarification.

Closing since this is a feature, not a bug 🤷‍♂
